h2/src/main/org/h2/value/Value.java

```diff
@@ -429,55 +429,55 @@ public static void clearCache() {
     }
 
     public boolean getBoolean() {
-        return ((ValueBoolean) convertTo(Value.BOOLEAN)).getBoolean();
+        return convertTo(Value.BOOLEAN).getBoolean();
     }
 
     public Date getDate() {
-        return ((ValueDate) convertTo(Value.DATE)).getDate();
+        return convertTo(Value.DATE).getDate();
     }
 
     public Time getTime() {
-        return ((ValueTime) convertTo(Value.TIME)).getTime();
+        return convertTo(Value.TIME).getTime();
     }
 
     public Timestamp getTimestamp() {
-        return ((ValueTimestamp) convertTo(Value.TIMESTAMP)).getTimestamp();
+        return convertTo(Value.TIMESTAMP).getTimestamp();
     }
 
     public byte[] getBytes() {
-        return ((ValueBytes) convertTo(Value.BYTES)).getBytes();
+        return convertTo(Value.BYTES).getBytes();
     }
 
     public byte[] getBytesNoCopy() {
-        return ((ValueBytes) convertTo(Value.BYTES)).getBytesNoCopy();
+        return convertTo(Value.BYTES).getBytesNoCopy();
     }
 
     public byte getByte() {
-        return ((ValueByte) convertTo(Value.BYTE)).getByte();
+        return convertTo(Value.BYTE).getByte();
     }
 
     public short getShort() {
-        return ((ValueShort) convertTo(Value.SHORT)).getShort();
+        return convertTo(Value.SHORT).getShort();
     }
 
     public BigDecimal getBigDecimal() {
-        return ((ValueDecimal) convertTo(Value.DECIMAL)).getBigDecimal();
+        return convertTo(Value.DECIMAL).getBigDecimal();
     }
 
     public double getDouble() {
-        return ((ValueDouble) convertTo(Value.DOUBLE)).getDouble();
+        return convertTo(Value.DOUBLE).getDouble();
     }
 
     public float getFloat() {
-        return ((ValueFloat) convertTo(Value.FLOAT)).getFloat();
+        return convertTo(Value.FLOAT).getFloat();
     }
 
     public int getInt() {
-        return ((ValueInt) convertTo(Value.INT)).getInt();
+        return convertTo(Value.INT).getInt();
     }
 
     public long getLong() {
-        return ((ValueLong) convertTo(Value.LONG)).getLong();
+        return convertTo(Value.LONG).getLong();
```

Comment:
All these typecasts look like a guards against invalid results of `convertTo()`.

After this change if result of `convertTo()` does not override corresponding getter behavior will be changed. We got `StackOverflowError` instead of `ClassCastException`. If some code expects possible `Exception', but not expects possible `Error` this may lead to a problem.

Hypothetically we should not have such broken `convertTo()` implementations, but anyway in looks like this change needs an additional analysis and review.